src/cb_admin/components/QrBox.js

```diff
@@ -25,27 +25,43 @@ const Button = styled(PrimaryButton) `
   font-size: ${fonts.size.small};
 `;
 
+const ErrorText = styled.p`
+  color: ${colors.error};
+`;
+
+const resendQrCodeState = {
```

Comment:
I’m unsure what you mean by this...

Reply:
The "implicit coupling" is in having to maintain two variable declarations in two different places (`resendQrCodeState` is defined in both components and if either definition changes these components wont work as expected).

I'm saying that there doesn't really appear to be any reason to send the status to this component. It is sufficient to send `hasSent` and `error`, since they hold the same amount of information.

As a side note, I'm not really happy with how the interface of this component was designed (by me, lol), but IMO it's not something to be dealt with in this PR.

Reply:
Gonna leave this for now, have opened #621 about a general solution.